src/site/stages/build/page-builder/transformers.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * A very specific helper function that expects to receive an
+ * array with one item which is an object with a single `value` property
+ *
+ */
+function arrayCollapse(arr) {
```

Comment:
If I understand it correctly, I think this function should be named differently to `takeLastValue` or something like that. When I read `arrayCollapse` I had thought that it meant something like [`flattenDeep`](https://lodash.com/docs/4.17.15#flattenDeep) but since it returns the array's last object's `value` property, we may want to find a more fitting name for future devs :)

Reply:
How do you feel about breaking this out into two functions?
1) Translate `{ value: 'anything' }` -> `'anything'`
2) Transform arrays into the most fitting data structure
```js
function transmogrifyArray(arr) {
  if (arr.length === 0) return null;
  if (arr.length === 1) return arr[0];
  return arr;
}
```

This way, if we have an array of `{ value: 'whatever' }` we'll end up with the array we expect while still not having to mess with `value`.

Reply:
I'm not convinced that breaking it out into two functions gets us anything.

@ElijahLynn may be able to weigh in, but it seems like the `[ { value: 'foo' } ]` pattern is a standard Drupal thing.  If that's the case, then what about something like:


```js

function getDrupalValue(arr) {
  if (arr.length === 0) return null;
  if (arr.length === 1) return arr[0].value;
}
```